### PR TITLE
Notification service for sending emails with spring mail

### PR DESCRIPTION
**Pull request by ChristianVazques**

### CODE REVIEW

File: src/main/java/se/knowit/bookitnotification/model/NotificationValidator.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+public class NotificationValidator {
+
+    public Notification validateNotificationOrElseThrowException(Notification notification) {
+        validateEventId(notification.getEventId());
+        validateParticipant(notification.getParticipant());
+        return notification;
+    }
+
+    private void validateParticipant(Participant participant) {
```
Comment (allbiin) [inline src/main/java/se/knowit/bookitnotification/model/NotificationValidator.java:19]: Change order of methods so they match the order in which they are called i.e.

```
private void validateEventId(UUID eventId)
private void validateParticipant(Participant participant)
```

File: src/main/java/se/knowit/bookitnotification/service/NotificationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        try {
+            sendRegistrationNotification(notification);
+        } catch(IllegalArgumentException | MailException e) {
+            //TODO: log exception.
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void sendRegistrationNotification(Notification notification) throws IllegalArgumentException, MailException {
```
Comment (allbiin) [inline src/main/java/se/knowit/bookitnotification/service/NotificationServiceImpl.java]: Neither IllegalArgumentException or MailException are checked exception and therefore they don't need to be declared here.

**Comment by allbiin** [comment]
I think it would be nice if the service was more general for notifications i.e. that the caller just sends an object with a receiver, a subject, and content. I think it should be up to the service that wants to send the notification to define the notification and that the notification service only responsibilty should be to send the notification. What are your thoughts?

**Comment by ChristianVazques** [comment]
@allbiin That was the initial thought when I design the service. But I then thought of maybe having html templates for each type of email(create event, event registration) that would be used to construct the email. In that case I would probably store them in the same project. But that could also be done outside the service, and let the service consume a request with recipient,subject, body etc, like you said.

**Review by olamat — APPROVED** [review]

**Pull request merged**